rust-runtime/aws-smithy-client/src/retry.rs

```diff
@@ -315,12 +307,17 @@ where
         req: &Operation<Handler, R>,
         result: Result<&SdkSuccess<T>, &SdkError<E>>,
     ) -> Option<Self::Future> {
-        let sleep = match &self.sleep {
-            Some(sleep) => sleep,
-            None => return None,
-        };
         let policy = req.retry_policy();
         let retry = policy.classify(result);
+        let sleep = match &self.sleep_impl {
+            Some(sleep) => sleep,
+            None => {
+                if retry != RetryKind::NotRetryable {
+                    tracing::debug!("cannot retry because no sleep implementation exists");
```

Comment:
should this be a `warn` instead? or would that just be annoying for people?

Reply:
This will be called on every retry, so it should be debug to avoid spamming the logs. The client construction will log a warning, so we're covered.